src/shipit_static_analysis/shipit_static_analysis/report/mozreview.py

```diff
@@ -207,14 +264,20 @@ def comment(self, filename, first_line, num_lines, text,
             return
         translated_line_num = self.translate_line_num(filename, first_line)
 
-        data = {
+        comment = {
             'filediff_id': f.id,
             'first_line': translated_line_num,
             'num_lines': num_lines,
             'text': text,
             'issue_opened': issue_opened,
         }
-        self.comments.append(data)
+
+        # Check this issue is not already published
+        if comment in self.existing_comments:
```

Comment:
I'm not too familiar with Python, but are we absolutely certain that `map in collection` will look for a value match (deep comparison?) and not a reference match? (Since the `comment` variable is declared just above, looking for a reference match would always fail.)

Reply:
It indeed compares the content of the dict, so it works (as shwocased i hope per the unit test)